hw-model/src/otp_provision.rs

```diff
@@ -691,6 +694,129 @@ pub fn otp_generate_manuf_debug_unlock_token_mem(
+}
+
+/// Generate the OTP memory contents for the SW_MANUF partition, including the digest.
+pub fn otp_generate_sw_manuf_partition_mem(
+    sw_manuf_partition: &OtpSwManufPartition,
+) -> Result<[u8; OTP_SW_MANUF_PARTITION_SIZE]> {
+    let mut output = [0u8; OTP_SW_MANUF_PARTITION_SIZE];
+    let mut offset = 0;
+
+    // Anti-Rollback Disable field.
+    let mut len = sw_manuf_partition.anti_rollback_disable.to_le_bytes().len();
+    output[offset..offset + len]
+        .copy_from_slice(&sw_manuf_partition.anti_rollback_disable.to_le_bytes());
+    offset += len;
+
+    // IDevID Cert Attributes field.
+    len = sw_manuf_partition.idevid_cert_attr.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.idevid_cert_attr);
+    offset += len;
+
+    // IDevID Cert field.
+    len = sw_manuf_partition.idevid_cert.to_le_bytes().len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.idevid_cert.to_le_bytes());
+    offset += len;
+
+    // HSM ID field.
+    len = sw_manuf_partition.hsm_id.to_le_bytes().len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.hsm_id.to_le_bytes());
+    offset += len;
+
+    // Stepping ID field.
+    len = sw_manuf_partition.stepping_id.to_le_bytes().len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.stepping_id.to_le_bytes());
+    offset += len;
+
+    // Prod debug unlock public key hash fields.
+    len = sw_manuf_partition.prod_debug_unlock_pks_0.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_0);
+    offset += len;
+    len = sw_manuf_partition.prod_debug_unlock_pks_1.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_1);
+    offset += len;
+    len = sw_manuf_partition.prod_debug_unlock_pks_2.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_2);
+    offset += len;
+    len = sw_manuf_partition.prod_debug_unlock_pks_3.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_3);
+    offset += len;
+    len = sw_manuf_partition.prod_debug_unlock_pks_4.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_4);
+    offset += len;
+    len = sw_manuf_partition.prod_debug_unlock_pks_5.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_5);
+    offset += len;
+    len = sw_manuf_partition.prod_debug_unlock_pks_6.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_6);
+    offset += len;
+    len = sw_manuf_partition.prod_debug_unlock_pks_7.len();
+    output[offset..offset + len].copy_from_slice(&sw_manuf_partition.prod_debug_unlock_pks_7);
+    offset += len;
+
+    // Compute and write digest field to lock the partition.
+    output[offset..OTP_SW_MANUF_PARTITION_SIZE - DIGEST_SIZE].fill(0);
```

Comment:
You should be able to skip this step because `output` was defined zeroized

Reply:
good catch. fixed.